Sources/generate-swift-syntax-builder/Templates/BuildableNodesFile.swift

```diff
@@ -277,12 +277,18 @@ private func createBuildFunction(node: Node) -> FunctionDecl {
         }
       }
     )
-    ReturnStmt(expression: FunctionCallExpr(MemberAccessExpr(base: "result", name: "withLeadingTrivia")) {
-      TupleExprElement(expression: FunctionCallExpr(MemberAccessExpr(
-        base: "combinedLeadingTrivia",
-        name: "addingSpacingAfterNewlinesIfNeeded"
-      )))
-    })
+    IfStmt(
+      conditions: ExprList([MemberAccessExpr(base: "combinedLeadingTrivia", name: "isEmpty")])
+    ) {
+      ReturnStmt(expression: "result")
+    } elseBody: {
+      ReturnStmt(expression: FunctionCallExpr(MemberAccessExpr(base: "result", name: "withLeadingTrivia")) {
+        TupleExprElement(expression: FunctionCallExpr(MemberAccessExpr(
+          base: "combinedLeadingTrivia",
+          name: "addingSpacingAfterNewlinesIfNeeded"
+        )))
+      })
+    }
```

Comment:
One idea might be to move this logic into `withLeadingTrivia` itself (and similarly into `withTrailingTrivia`). However since we cannot assume that the syntax doesn't already have trivia (like in this case) the check would likely get more expensive, so I am not sure whether the more general solution would be worth it there.